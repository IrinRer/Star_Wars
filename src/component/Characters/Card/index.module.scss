.wrapper {
  display: flex;
  flex-wrap: wrap;
  gap: 40px;
  margin-top: 100px;
  justify-content: center;
  .wrapper_card {
    filter: drop-shadow(4px 4px 4px rgba(33, 33, 33, 0.1));
    background: #f0f0f0;
    border-radius: 8px;
    width: 300px;
    cursor: pointer;
    padding: 5px 20px;

    .title {
      font-weight: 700;
      font-size: 18px;
      line-height: 21px;
      margin: 0;
      text-shadow: 4px 4px 4px rgba(33, 33, 33, 0.1);
    }
  }
}

.tag_wrapper {
    display: flex;
    gap: 20px;
    margin-top: 15px;
}
