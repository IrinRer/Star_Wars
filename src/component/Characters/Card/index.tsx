import Loader from 'component/Loader';
import Modal from 'component/Modal';
import { useAppDispatch } from 'hooks/redux/useAppDispatch';
import { useAppSelector } from 'hooks/redux/useAppSelector';
import React, { useState } from 'react';
import {
  getCharacters,
  getLoading,
  getSearchLoading,
} from 'store/data/selectors';
import { setSelectedCharacter } from 'store/data/slice';
import Circle from '../Circle';
import Tag from '../Tag';
import styles from './index.module.scss';

const Card = () => {
  const characters = useAppSelector(getCharacters);
  const loading = useAppSelector(getLoading);
  const loadingSearch = useAppSelector(getSearchLoading);
  const dispatch = useAppDispatch();
  const [isOpen, setOpen] = useState<boolean>(false);

  const handleClick = (character) => {
    setOpen((state) => !state);
    dispatch(setSelectedCharacter(character));
  };

  return (
    <div className={styles.wrapper}>
      {!loadingSearch && !loading ? (
        characters.map((item) => (
          <div
            className={styles.wrapper_card}
            onClick={() => handleClick(item)}
            key={item.name}
          >
            <p className={styles.title}>{item.name}</p>
            <Circle height={item.height} mass={item.mass} block={false} />
            <div className={styles.tag_wrapper}>
              <Tag text={item.gender} />
              <Tag text={item.birth_year} />
            </div>
          </div>
        ))
      ) : (
        <Loader dark />
      )}
      <Modal isOpen={isOpen} setOpen={setOpen} />
    </div>
  );
};

export default Card;
