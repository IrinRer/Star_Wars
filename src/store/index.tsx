import { configureStore } from '@reduxjs/toolkit';
import  DataReducer from './data/slice';

export const store = configureStore({
  reducer: {
    data: DataReducer
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type RootState = {
    data: any;
};

export type AppDispatch = typeof store.dispatch;
