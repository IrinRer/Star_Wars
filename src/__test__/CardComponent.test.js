import React from 'react';
import axios from 'axios';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../store/index';
import CharactersComponent from '../component/Characters/index';
import Card from '../component/Characters/Card';

describe('CharactersComponent component', () => {
  it('CharactersComponent render"', () => {
    render(
      <Provider store={store}>
        <CharactersComponent />
      </Provider>,
    );

    expect(screen.getByText('favorite')).toBeInTheDocument();
  });

  it('CharactersComponent snapshot', () => {
    const card = render(
      <Provider store={store}>
        <CharactersComponent />
      </Provider>,
    );

    expect(card).toMatchSnapshot();
  });
});

describe('Card component', () => {
  test('it displays a list of people-list', async () => {
    render(
      <Provider store={store}>
        <Card />
      </Provider>,
    );

    const peopleList = await waitFor(() => screen.getByTestId('people-list'));
    expect(peopleList).toBeInTheDocument();
  });
});

jest.mock('axios');

const fakeData = {
  name: 'test name',
  eye_color: 'test color eye',
  skin_color: 'test color skin',
};

axios.get.mockResolvedValue({ data: fakeData });

describe('Card component', () => {
  test('it displays a row for each people', async () => {
    axios.get.mockResolvedValue({ data: fakeData });

    render(
      <Provider store={store}>
        <Card />
      </Provider>,
    );

    const peopleList = await waitFor(() =>
      screen.findAllByTestId('people-list'),
    );
    expect(peopleList).toHaveLength(1);
  });
});
