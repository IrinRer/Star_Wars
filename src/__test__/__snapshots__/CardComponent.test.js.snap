// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersComponent component CharactersComponent snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
        data-testid="wrapper"
      >
        <div>
          <p
            class="title"
          >
            <b>
              0
               Peoples
            </b>
             for you to choose your 
            <b>
              favorite
            </b>
          </p>
          <input
            class="input"
            placeholder="find..."
            type="text"
            value=""
          />
        </div>
        <div
          class="wrapper_main"
        >
          <div
            class="wrapper"
            data-testid="people-list"
          >
            <svg>
              spinner_dark.svg
            </svg>
          </div>
          <div
            class="pagination"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
      data-testid="wrapper"
    >
      <div>
        <p
          class="title"
        >
          <b>
            0
             Peoples
          </b>
           for you to choose your 
          <b>
            favorite
          </b>
        </p>
        <input
          class="input"
          placeholder="find..."
          type="text"
          value=""
        />
      </div>
      <div
        class="wrapper_main"
      >
        <div
          class="wrapper"
          data-testid="people-list"
        >
          <svg>
            spinner_dark.svg
          </svg>
        </div>
        <div
          class="pagination"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
